docarray/proto/docarray.proto

```diff
@@ -46,6 +46,15 @@ message NodeProto {
 
     // a sub DocumentArray
     DocumentArrayProto chunks = 5;
+
+    NdArrayProto embedding = 6;
+
+    string any_url = 7;
+
+    string image_url = 8;
+
+    string id = 9;
```

Comment:
Id should not be a oneof, it should be an attribute next to the map

Reply:
i am not convince. At the Document level the ID is in the dict as well

Reply:
let me think about it